modules/solid_mechanics/src/timeintegrators/ExplicitMixedOrder.C

```diff
@@ -87,6 +92,95 @@ ExplicitMixedOrder::ExplicitMixedOrder(const InputParameters & parameters)
   _fe_problem.setUDotRequested(true);
   _fe_problem.setUDotOldRequested(true);
   _fe_problem.setUDotDotRequested(true);
+
+  // This effectively changes the default solve_type to LINEAR instead of PJFNK,
+  // so that it is valid to not supply solve_type in the Executioner block:
+  if (_nl)
+    _fe_problem.solverParams(_nl->number())._type = Moose::ST_LINEAR;
+
+  _ones = addVector("ones", true, PARALLEL);
+
+  // don't set any of the common SNES-related petsc options to prevent unused option warnings
+  Moose::PetscSupport::dontAddCommonSNESOptions(_fe_problem);
+}
+
+void
+ExplicitMixedOrder::meshChanged()
+{
+  *_ones = 1.;
+}
+
+void
+ExplicitMixedOrder::init()
```

Comment:
This function is very long and almost identical to `ExplicitTimeIntegrator::init`. Can we somehow reduce code duplication?